waveorder/util.py

```diff
@@ -2239,3 +2242,67 @@ def orientation_3D_continuity_map(
     retardance_pr_avg /= np.max(retardance_pr_avg)
 
     return retardance_pr_avg
+
+
+def pauli():
+    # yx order
+    # trace-orthogonal normalization
+    # torch.einsum("kij,lji->kl", pauli(), pauli()) == torch.eye(4)
+
+    # intensity, x-y, +45-(-45), LCP-RCP
+    # yx
+    # yx
+    a = 2**-0.5
+    sigma = torch.tensor(
+        [
+            [[a, 0], [0, a]],
+            [[-a, 0], [0, a]],
+            [[0, a], [a, 0]],
+            [[0, 1j * a], [-1j * a, 0]],
+        ]
+    )
+    return sigma
+
+
+# torch.allclose(
+#     torch.abs(torch.einsum("kij,lji->kl", s, s) - torch.eye(4)),
+#     torch.zeros((4, 4)),
+#     atol=1e-5,
+# )
+
+
+def gellmann():
+    # zyx order
+    # trace-orthogonal normalization
+    # torch.einsum("kij,lji->kl", gellmann(), gellmann()) == torch.eye(9)
+    #
+    # lexicographical order of the Gell-Mann matrices
+    # 00, 1-1, 10, 11, 2-2, 2-1, 20, 21, 22
+    #
+    # zyx
+    # zyx
+    a = 3**-0.5
+    c = 2**-0.5
+    d = -(6**-0.5)
+    e = 2 * (6**-0.5)
+    return torch.tensor(
+        [
+            [[a, 0, 0], [0, a, 0], [0, 0, a]],
+            [[0, 0, -c], [0, 0, 0], [c, 0, 0]],
+            [[0, 0, 0], [0, 0, -c], [0, c, 0]],
+            [[0, -c, 0], [c, 0, 0], [0, 0, 0]],
+            [[0, 0, 0], [0, 0, c], [0, c, 0]],  #
+            [[0, c, 0], [c, 0, 0], [0, 0, 0]],
+            [[e, 0, 0], [0, d, 0], [0, 0, d]],
+            [[0, 0, c], [0, 0, 0], [c, 0, 0]],
+            [[0, 0, 0], [0, -c, 0], [0, 0, c]],  #
+        ], dtype=torch.complex64
+    )
+
+    # torch.allclose(
+
+
+#     torch.abs(torch.einsum("kij,lji->kl", Y, Y) - torch.eye(9)),
+#     torch.zeros((9, 9)),
+#     atol=1e-5,
+# )
```

Comment:
```suggestion
```
move to test?

Reply:
Thanks for flagging! I meant to do this and forgot. Moved to tests.